R/get_osw_query.R

```diff
@@ -254,8 +316,33 @@ getPrecursorsQuery <- function(runType = "DIA_Proteomics", level = "Peptide")
       WHERE PRECURSOR.DECOY = 0
       ORDER BY peptide_id, transition_group_id, transition_id;"
     }
-  } else if (runType == "DIA_Metabolomics")
-  {
+  } else if (runType == "DIA_IPF"){
+    ## TODO: Not as clear on PROTEIN level, for now we only do peptide level
+      query <- "SELECT DISTINCT PRECURSOR.ID AS transition_group_id,
+      TRANSITION_PRECURSOR_MAPPING.TRANSITION_ID AS transition_id,
+      PEPTIDE.ID AS peptide_id,
+      PEPTIDE.MODIFIED_SEQUENCE AS sequence,
+      PRECURSOR.CHARGE AS charge,
+      PRECURSOR.GROUP_LABEL AS group_label
+      --TRANSITION.DETECTING AS detecting,
+	    --TRANSITION.IDENTIFYING AS identifying
+      FROM PRECURSOR
+      INNER JOIN TRANSITION_PRECURSOR_MAPPING ON TRANSITION_PRECURSOR_MAPPING.PRECURSOR_ID = PRECURSOR.ID
+      INNER JOIN TRANSITION ON TRANSITION.ID = TRANSITION_PRECURSOR_MAPPING.TRANSITION_ID
```

Comment:
It will be faster if you select IDENTIFYING and DETECTING columns before INNER JOIN

Reply:
So you mean before inner joining the `TRANISITION` table, to create a smaller sub-table to select the `IDENTIFYING` and `DETECTING`, and then use the smaller sub-table to inner join? Makes sense

Reply:
Yes. It would matter here because TRANSITION table has a lot of columns.